lib/src/facts/posix/networking_resolver.cc

```diff
@@ -120,4 +120,32 @@ namespace facter { namespace facts { namespace posix {
         return result;
     }
 
+    void networking_resolver::assign_address(ipaddress& ip, sockaddr const* address, sockaddr const* mask) const
+    {
+        if (!address) {
+           return;
+        }
+
+        string* assigned = nullptr;
+        bool (*ignored)(string const&) = nullptr;
+
+        // Check for IPv4 or IPv6 addresses
+        if (address->sa_family == AF_INET) {
+            assigned = &ip.v4;
+            ignored = ignored_ipv4_address;
+        } else if (address->sa_family == AF_INET6) {
+            assigned = &ip.v6;
+            ignored = ignored_ipv6_address;
+        } else {
+            return;
+        }
+
+        // If the current address is not ignored or there is existing ignored address, do not assign
+        if (!ignored(*assigned) || !assigned->empty()) {
```

Comment:
If `ignored` returns true, then it could be a non-empty ignored address.  This second check ensures that we do not overwrite an existing non-empty ignored address with another one (this is FACT-1147).